// @aws-sdk/client-wafv2/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');
var schemas_0 = require('./schemas/schemas_0');
var errors = require('./models/errors');
var WAFV2ServiceException = require('./models/WAFV2ServiceException');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "wafv2",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class WAFV2Client extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultWAFV2HttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class AssociateWebACLCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "AssociateWebACL", {})
    .n("WAFV2Client", "AssociateWebACLCommand")
    .sc(schemas_0.AssociateWebACL$)
    .build() {
}

class CheckCapacityCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "CheckCapacity", {})
    .n("WAFV2Client", "CheckCapacityCommand")
    .sc(schemas_0.CheckCapacity$)
    .build() {
}

class CreateAPIKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "CreateAPIKey", {})
    .n("WAFV2Client", "CreateAPIKeyCommand")
    .sc(schemas_0.CreateAPIKey$)
    .build() {
}

class CreateIPSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "CreateIPSet", {})
    .n("WAFV2Client", "CreateIPSetCommand")
    .sc(schemas_0.CreateIPSet$)
    .build() {
}

class CreateRegexPatternSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "CreateRegexPatternSet", {})
    .n("WAFV2Client", "CreateRegexPatternSetCommand")
    .sc(schemas_0.CreateRegexPatternSet$)
    .build() {
}

class CreateRuleGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "CreateRuleGroup", {})
    .n("WAFV2Client", "CreateRuleGroupCommand")
    .sc(schemas_0.CreateRuleGroup$)
    .build() {
}

class CreateWebACLCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "CreateWebACL", {})
    .n("WAFV2Client", "CreateWebACLCommand")
    .sc(schemas_0.CreateWebACL$)
    .build() {
}

class DeleteAPIKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "DeleteAPIKey", {})
    .n("WAFV2Client", "DeleteAPIKeyCommand")
    .sc(schemas_0.DeleteAPIKey$)
    .build() {
}

class DeleteFirewallManagerRuleGroupsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "DeleteFirewallManagerRuleGroups", {})
    .n("WAFV2Client", "DeleteFirewallManagerRuleGroupsCommand")
    .sc(schemas_0.DeleteFirewallManagerRuleGroups$)
    .build() {
}

class DeleteIPSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "DeleteIPSet", {})
    .n("WAFV2Client", "DeleteIPSetCommand")
    .sc(schemas_0.DeleteIPSet$)
    .build() {
}

class DeleteLoggingConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "DeleteLoggingConfiguration", {})
    .n("WAFV2Client", "DeleteLoggingConfigurationCommand")
    .sc(schemas_0.DeleteLoggingConfiguration$)
    .build() {
}

class DeletePermissionPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "DeletePermissionPolicy", {})
    .n("WAFV2Client", "DeletePermissionPolicyCommand")
    .sc(schemas_0.DeletePermissionPolicy$)
    .build() {
}

class DeleteRegexPatternSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "DeleteRegexPatternSet", {})
    .n("WAFV2Client", "DeleteRegexPatternSetCommand")
    .sc(schemas_0.DeleteRegexPatternSet$)
    .build() {
}

class DeleteRuleGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "DeleteRuleGroup", {})
    .n("WAFV2Client", "DeleteRuleGroupCommand")
    .sc(schemas_0.DeleteRuleGroup$)
    .build() {
}

class DeleteWebACLCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "DeleteWebACL", {})
    .n("WAFV2Client", "DeleteWebACLCommand")
    .sc(schemas_0.DeleteWebACL$)
    .build() {
}

class DescribeAllManagedProductsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "DescribeAllManagedProducts", {})
    .n("WAFV2Client", "DescribeAllManagedProductsCommand")
    .sc(schemas_0.DescribeAllManagedProducts$)
    .build() {
}

class DescribeManagedProductsByVendorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "DescribeManagedProductsByVendor", {})
    .n("WAFV2Client", "DescribeManagedProductsByVendorCommand")
    .sc(schemas_0.DescribeManagedProductsByVendor$)
    .build() {
}

class DescribeManagedRuleGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "DescribeManagedRuleGroup", {})
    .n("WAFV2Client", "DescribeManagedRuleGroupCommand")
    .sc(schemas_0.DescribeManagedRuleGroup$)
    .build() {
}

class DisassociateWebACLCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "DisassociateWebACL", {})
    .n("WAFV2Client", "DisassociateWebACLCommand")
    .sc(schemas_0.DisassociateWebACL$)
    .build() {
}

class GenerateMobileSdkReleaseUrlCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "GenerateMobileSdkReleaseUrl", {})
    .n("WAFV2Client", "GenerateMobileSdkReleaseUrlCommand")
    .sc(schemas_0.GenerateMobileSdkReleaseUrl$)
    .build() {
}

class GetDecryptedAPIKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "GetDecryptedAPIKey", {})
    .n("WAFV2Client", "GetDecryptedAPIKeyCommand")
    .sc(schemas_0.GetDecryptedAPIKey$)
    .build() {
}

class GetIPSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "GetIPSet", {})
    .n("WAFV2Client", "GetIPSetCommand")
    .sc(schemas_0.GetIPSet$)
    .build() {
}

class GetLoggingConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "GetLoggingConfiguration", {})
    .n("WAFV2Client", "GetLoggingConfigurationCommand")
    .sc(schemas_0.GetLoggingConfiguration$)
    .build() {
}

class GetManagedRuleSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "GetManagedRuleSet", {})
    .n("WAFV2Client", "GetManagedRuleSetCommand")
    .sc(schemas_0.GetManagedRuleSet$)
    .build() {
}

class GetMobileSdkReleaseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "GetMobileSdkRelease", {})
    .n("WAFV2Client", "GetMobileSdkReleaseCommand")
    .sc(schemas_0.GetMobileSdkRelease$)
    .build() {
}

class GetPermissionPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "GetPermissionPolicy", {})
    .n("WAFV2Client", "GetPermissionPolicyCommand")
    .sc(schemas_0.GetPermissionPolicy$)
    .build() {
}

class GetRateBasedStatementManagedKeysCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "GetRateBasedStatementManagedKeys", {})
    .n("WAFV2Client", "GetRateBasedStatementManagedKeysCommand")
    .sc(schemas_0.GetRateBasedStatementManagedKeys$)
    .build() {
}

class GetRegexPatternSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "GetRegexPatternSet", {})
    .n("WAFV2Client", "GetRegexPatternSetCommand")
    .sc(schemas_0.GetRegexPatternSet$)
    .build() {
}

class GetRuleGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "GetRuleGroup", {})
    .n("WAFV2Client", "GetRuleGroupCommand")
    .sc(schemas_0.GetRuleGroup$)
    .build() {
}

class GetSampledRequestsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "GetSampledRequests", {})
    .n("WAFV2Client", "GetSampledRequestsCommand")
    .sc(schemas_0.GetSampledRequests$)
    .build() {
}

class GetTopPathStatisticsByTrafficCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "GetTopPathStatisticsByTraffic", {})
    .n("WAFV2Client", "GetTopPathStatisticsByTrafficCommand")
    .sc(schemas_0.GetTopPathStatisticsByTraffic$)
    .build() {
}

class GetWebACLCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "GetWebACL", {})
    .n("WAFV2Client", "GetWebACLCommand")
    .sc(schemas_0.GetWebACL$)
    .build() {
}

class GetWebACLForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "GetWebACLForResource", {})
    .n("WAFV2Client", "GetWebACLForResourceCommand")
    .sc(schemas_0.GetWebACLForResource$)
    .build() {
}

class ListAPIKeysCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "ListAPIKeys", {})
    .n("WAFV2Client", "ListAPIKeysCommand")
    .sc(schemas_0.ListAPIKeys$)
    .build() {
}

class ListAvailableManagedRuleGroupsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "ListAvailableManagedRuleGroups", {})
    .n("WAFV2Client", "ListAvailableManagedRuleGroupsCommand")
    .sc(schemas_0.ListAvailableManagedRuleGroups$)
    .build() {
}

class ListAvailableManagedRuleGroupVersionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "ListAvailableManagedRuleGroupVersions", {})
    .n("WAFV2Client", "ListAvailableManagedRuleGroupVersionsCommand")
    .sc(schemas_0.ListAvailableManagedRuleGroupVersions$)
    .build() {
}

class ListIPSetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "ListIPSets", {})
    .n("WAFV2Client", "ListIPSetsCommand")
    .sc(schemas_0.ListIPSets$)
    .build() {
}

class ListLoggingConfigurationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "ListLoggingConfigurations", {})
    .n("WAFV2Client", "ListLoggingConfigurationsCommand")
    .sc(schemas_0.ListLoggingConfigurations$)
    .build() {
}

class ListManagedRuleSetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "ListManagedRuleSets", {})
    .n("WAFV2Client", "ListManagedRuleSetsCommand")
    .sc(schemas_0.ListManagedRuleSets$)
    .build() {
}

class ListMobileSdkReleasesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "ListMobileSdkReleases", {})
    .n("WAFV2Client", "ListMobileSdkReleasesCommand")
    .sc(schemas_0.ListMobileSdkReleases$)
    .build() {
}

class ListRegexPatternSetsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "ListRegexPatternSets", {})
    .n("WAFV2Client", "ListRegexPatternSetsCommand")
    .sc(schemas_0.ListRegexPatternSets$)
    .build() {
}

class ListResourcesForWebACLCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "ListResourcesForWebACL", {})
    .n("WAFV2Client", "ListResourcesForWebACLCommand")
    .sc(schemas_0.ListResourcesForWebACL$)
    .build() {
}

class ListRuleGroupsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "ListRuleGroups", {})
    .n("WAFV2Client", "ListRuleGroupsCommand")
    .sc(schemas_0.ListRuleGroups$)
    .build() {
}

class ListTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "ListTagsForResource", {})
    .n("WAFV2Client", "ListTagsForResourceCommand")
    .sc(schemas_0.ListTagsForResource$)
    .build() {
}

class ListWebACLsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "ListWebACLs", {})
    .n("WAFV2Client", "ListWebACLsCommand")
    .sc(schemas_0.ListWebACLs$)
    .build() {
}

class PutLoggingConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "PutLoggingConfiguration", {})
    .n("WAFV2Client", "PutLoggingConfigurationCommand")
    .sc(schemas_0.PutLoggingConfiguration$)
    .build() {
}

class PutManagedRuleSetVersionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "PutManagedRuleSetVersions", {})
    .n("WAFV2Client", "PutManagedRuleSetVersionsCommand")
    .sc(schemas_0.PutManagedRuleSetVersions$)
    .build() {
}

class PutPermissionPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "PutPermissionPolicy", {})
    .n("WAFV2Client", "PutPermissionPolicyCommand")
    .sc(schemas_0.PutPermissionPolicy$)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "TagResource", {})
    .n("WAFV2Client", "TagResourceCommand")
    .sc(schemas_0.TagResource$)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "UntagResource", {})
    .n("WAFV2Client", "UntagResourceCommand")
    .sc(schemas_0.UntagResource$)
    .build() {
}

class UpdateIPSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "UpdateIPSet", {})
    .n("WAFV2Client", "UpdateIPSetCommand")
    .sc(schemas_0.UpdateIPSet$)
    .build() {
}

class UpdateManagedRuleSetVersionExpiryDateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "UpdateManagedRuleSetVersionExpiryDate", {})
    .n("WAFV2Client", "UpdateManagedRuleSetVersionExpiryDateCommand")
    .sc(schemas_0.UpdateManagedRuleSetVersionExpiryDate$)
    .build() {
}

class UpdateRegexPatternSetCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "UpdateRegexPatternSet", {})
    .n("WAFV2Client", "UpdateRegexPatternSetCommand")
    .sc(schemas_0.UpdateRegexPatternSet$)
    .build() {
}

class UpdateRuleGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "UpdateRuleGroup", {})
    .n("WAFV2Client", "UpdateRuleGroupCommand")
    .sc(schemas_0.UpdateRuleGroup$)
    .build() {
}

class UpdateWebACLCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AWSWAF_20190729", "UpdateWebACL", {})
    .n("WAFV2Client", "UpdateWebACLCommand")
    .sc(schemas_0.UpdateWebACL$)
    .build() {
}

const commands = {
    AssociateWebACLCommand,
    CheckCapacityCommand,
    CreateAPIKeyCommand,
    CreateIPSetCommand,
    CreateRegexPatternSetCommand,
    CreateRuleGroupCommand,
    CreateWebACLCommand,
    DeleteAPIKeyCommand,
    DeleteFirewallManagerRuleGroupsCommand,
    DeleteIPSetCommand,
    DeleteLoggingConfigurationCommand,
    DeletePermissionPolicyCommand,
    DeleteRegexPatternSetCommand,
    DeleteRuleGroupCommand,
    DeleteWebACLCommand,
    DescribeAllManagedProductsCommand,
    DescribeManagedProductsByVendorCommand,
    DescribeManagedRuleGroupCommand,
    DisassociateWebACLCommand,
    GenerateMobileSdkReleaseUrlCommand,
    GetDecryptedAPIKeyCommand,
    GetIPSetCommand,
    GetLoggingConfigurationCommand,
    GetManagedRuleSetCommand,
    GetMobileSdkReleaseCommand,
    GetPermissionPolicyCommand,
    GetRateBasedStatementManagedKeysCommand,
    GetRegexPatternSetCommand,
    GetRuleGroupCommand,
    GetSampledRequestsCommand,
    GetTopPathStatisticsByTrafficCommand,
    GetWebACLCommand,
    GetWebACLForResourceCommand,
    ListAPIKeysCommand,
    ListAvailableManagedRuleGroupsCommand,
    ListAvailableManagedRuleGroupVersionsCommand,
    ListIPSetsCommand,
    ListLoggingConfigurationsCommand,
    ListManagedRuleSetsCommand,
    ListMobileSdkReleasesCommand,
    ListRegexPatternSetsCommand,
    ListResourcesForWebACLCommand,
    ListRuleGroupsCommand,
    ListTagsForResourceCommand,
    ListWebACLsCommand,
    PutLoggingConfigurationCommand,
    PutManagedRuleSetVersionsCommand,
    PutPermissionPolicyCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateIPSetCommand,
    UpdateManagedRuleSetVersionExpiryDateCommand,
    UpdateRegexPatternSetCommand,
    UpdateRuleGroupCommand,
    UpdateWebACLCommand,
};
class WAFV2 extends WAFV2Client {
}
smithyClient.createAggregatedClient(commands, WAFV2);

const ActionValue = {
    ALLOW: "ALLOW",
    BLOCK: "BLOCK",
    CAPTCHA: "CAPTCHA",
    CHALLENGE: "CHALLENGE",
    COUNT: "COUNT",
    EXCLUDED_AS_COUNT: "EXCLUDED_AS_COUNT",
};
const FallbackBehavior = {
    MATCH: "MATCH",
    NO_MATCH: "NO_MATCH",
};
const OversizeHandling = {
    CONTINUE: "CONTINUE",
    MATCH: "MATCH",
    NO_MATCH: "NO_MATCH",
};
const MapMatchScope = {
    ALL: "ALL",
    KEY: "KEY",
    VALUE: "VALUE",
};
const BodyParsingFallbackBehavior = {
    EVALUATE_AS_STRING: "EVALUATE_AS_STRING",
    MATCH: "MATCH",
    NO_MATCH: "NO_MATCH",
};
const JsonMatchScope = {
    ALL: "ALL",
    KEY: "KEY",
    VALUE: "VALUE",
};
const PositionalConstraint = {
    CONTAINS: "CONTAINS",
    CONTAINS_WORD: "CONTAINS_WORD",
    ENDS_WITH: "ENDS_WITH",
    EXACTLY: "EXACTLY",
    STARTS_WITH: "STARTS_WITH",
};
const TextTransformationType = {
    BASE64_DECODE: "BASE64_DECODE",
    BASE64_DECODE_EXT: "BASE64_DECODE_EXT",
    CMD_LINE: "CMD_LINE",
    COMPRESS_WHITE_SPACE: "COMPRESS_WHITE_SPACE",
    CSS_DECODE: "CSS_DECODE",
    ESCAPE_SEQ_DECODE: "ESCAPE_SEQ_DECODE",
    HEX_DECODE: "HEX_DECODE",
    HTML_ENTITY_DECODE: "HTML_ENTITY_DECODE",
    JS_DECODE: "JS_DECODE",
    LOWERCASE: "LOWERCASE",
    MD5: "MD5",
    NONE: "NONE",
    NORMALIZE_PATH: "NORMALIZE_PATH",
    NORMALIZE_PATH_WIN: "NORMALIZE_PATH_WIN",
    REMOVE_NULLS: "REMOVE_NULLS",
    REPLACE_COMMENTS: "REPLACE_COMMENTS",
    REPLACE_NULLS: "REPLACE_NULLS",
    SQL_HEX_DECODE: "SQL_HEX_DECODE",
    URL_DECODE: "URL_DECODE",
    URL_DECODE_UNI: "URL_DECODE_UNI",
    UTF8_TO_UNICODE: "UTF8_TO_UNICODE",
};
const CountryCode = {
    AD: "AD",
    AE: "AE",
    AF: "AF",
    AG: "AG",
    AI: "AI",
    AL: "AL",
    AM: "AM",
    AO: "AO",
    AQ: "AQ",
    AR: "AR",
    AS: "AS",
    AT: "AT",
    AU: "AU",
    AW: "AW",
    AX: "AX",
    AZ: "AZ",
    BA: "BA",
    BB: "BB",
    BD: "BD",
    BE: "BE",
    BF: "BF",
    BG: "BG",
    BH: "BH",
    BI: "BI",
    BJ: "BJ",
    BL: "BL",
    BM: "BM",
    BN: "BN",
    BO: "BO",
    BQ: "BQ",
    BR: "BR",
    BS: "BS",
    BT: "BT",
    BV: "BV",
    BW: "BW",
    BY: "BY",
    BZ: "BZ",
    CA: "CA",
    CC: "CC",
    CD: "CD",
    CF: "CF",
    CG: "CG",
    CH: "CH",
    CI: "CI",
    CK: "CK",
    CL: "CL",
    CM: "CM",
    CN: "CN",
    CO: "CO",
    CR: "CR",
    CU: "CU",
    CV: "CV",
    CW: "CW",
    CX: "CX",
    CY: "CY",
    CZ: "CZ",
    DE: "DE",
    DJ: "DJ",
    DK: "DK",
    DM: "DM",
    DO: "DO",
    DZ: "DZ",
    EC: "EC",
    EE: "EE",
    EG: "EG",
    EH: "EH",
    ER: "ER",
    ES: "ES",
    ET: "ET",
    FI: "FI",
    FJ: "FJ",
    FK: "FK",
    FM: "FM",
    FO: "FO",
    FR: "FR",
    GA: "GA",
    GB: "GB",
    GD: "GD",
    GE: "GE",
    GF: "GF",
    GG: "GG",
    GH: "GH",
    GI: "GI",
    GL: "GL",
    GM: "GM",
    GN: "GN",
    GP: "GP",
    GQ: "GQ",
    GR: "GR",
    GS: "GS",
    GT: "GT",
    GU: "GU",
    GW: "GW",
    GY: "GY",
    HK: "HK",
    HM: "HM",
    HN: "HN",
    HR: "HR",
    HT: "HT",
    HU: "HU",
    ID: "ID",
    IE: "IE",
    IL: "IL",
    IM: "IM",
    IN: "IN",
    IO: "IO",
    IQ: "IQ",
    IR: "IR",
    IS: "IS",
    IT: "IT",
    JE: "JE",
    JM: "JM",
    JO: "JO",
    JP: "JP",
    KE: "KE",
    KG: "KG",
    KH: "KH",
    KI: "KI",
    KM: "KM",
    KN: "KN",
    KP: "KP",
    KR: "KR",
    KW: "KW",
    KY: "KY",
    KZ: "KZ",
    LA: "LA",
    LB: "LB",
    LC: "LC",
    LI: "LI",
    LK: "LK",
    LR: "LR",
    LS: "LS",
    LT: "LT",
    LU: "LU",
    LV: "LV",
    LY: "LY",
    MA: "MA",
    MC: "MC",
    MD: "MD",
    ME: "ME",
    MF: "MF",
    MG: "MG",
    MH: "MH",
    MK: "MK",
    ML: "ML",
    MM: "MM",
    MN: "MN",
    MO: "MO",
    MP: "MP",
    MQ: "MQ",
    MR: "MR",
    MS: "MS",
    MT: "MT",
    MU: "MU",
    MV: "MV",
    MW: "MW",
    MX: "MX",
    MY: "MY",
    MZ: "MZ",
    NA: "NA",
    NC: "NC",
    NE: "NE",
    NF: "NF",
    NG: "NG",
    NI: "NI",
    NL: "NL",
    NO: "NO",
    NP: "NP",
    NR: "NR",
    NU: "NU",
    NZ: "NZ",
    OM: "OM",
    PA: "PA",
    PE: "PE",
    PF: "PF",
    PG: "PG",
    PH: "PH",
    PK: "PK",
    PL: "PL",
    PM: "PM",
    PN: "PN",
    PR: "PR",
    PS: "PS",
    PT: "PT",
    PW: "PW",
    PY: "PY",
    QA: "QA",
    RE: "RE",
    RO: "RO",
    RS: "RS",
    RU: "RU",
    RW: "RW",
    SA: "SA",
    SB: "SB",
    SC: "SC",
    SD: "SD",
    SE: "SE",
    SG: "SG",
    SH: "SH",
    SI: "SI",
    SJ: "SJ",
    SK: "SK",
    SL: "SL",
    SM: "SM",
    SN: "SN",
    SO: "SO",
    SR: "SR",
    SS: "SS",
    ST: "ST",
    SV: "SV",
    SX: "SX",
    SY: "SY",
    SZ: "SZ",
    TC: "TC",
    TD: "TD",
    TF: "TF",
    TG: "TG",
    TH: "TH",
    TJ: "TJ",
    TK: "TK",
    TL: "TL",
    TM: "TM",
    TN: "TN",
    TO: "TO",
    TR: "TR",
    TT: "TT",
    TV: "TV",
    TW: "TW",
    TZ: "TZ",
    UA: "UA",
    UG: "UG",
    UM: "UM",
    US: "US",
    UY: "UY",
    UZ: "UZ",
    VA: "VA",
    VC: "VC",
    VE: "VE",
    VG: "VG",
    VI: "VI",
    VN: "VN",
    VU: "VU",
    WF: "WF",
    WS: "WS",
    XK: "XK",
    YE: "YE",
    YT: "YT",
    ZA: "ZA",
    ZM: "ZM",
    ZW: "ZW",
};
const ForwardedIPPosition = {
    ANY: "ANY",
    FIRST: "FIRST",
    LAST: "LAST",
};
const LabelMatchScope = {
    LABEL: "LABEL",
    NAMESPACE: "NAMESPACE",
};
const PayloadType = {
    FORM_ENCODED: "FORM_ENCODED",
    JSON: "JSON",
};
const SensitivityToAct = {
    HIGH: "HIGH",
    LOW: "LOW",
    MEDIUM: "MEDIUM",
};
const UsageOfAction = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const InspectionLevel = {
    COMMON: "COMMON",
    TARGETED: "TARGETED",
};
const RateBasedStatementAggregateKeyType = {
    CONSTANT: "CONSTANT",
    CUSTOM_KEYS: "CUSTOM_KEYS",
    FORWARDED_IP: "FORWARDED_IP",
    IP: "IP",
};
const ComparisonOperator = {
    EQ: "EQ",
    GE: "GE",
    GT: "GT",
    LE: "LE",
    LT: "LT",
    NE: "NE",
};
const SensitivityLevel = {
    HIGH: "HIGH",
    LOW: "LOW",
};
const AssociatedResourceType = {
    API_GATEWAY: "API_GATEWAY",
    APP_RUNNER_SERVICE: "APP_RUNNER_SERVICE",
    CLOUDFRONT: "CLOUDFRONT",
    COGNITO_USER_POOL: "COGNITO_USER_POOL",
    VERIFIED_ACCESS_INSTANCE: "VERIFIED_ACCESS_INSTANCE",
};
const ParameterExceptionField = {
    ACP_RULE_SET_RESPONSE_INSPECTION: "ACP_RULE_SET_RESPONSE_INSPECTION",
    AND_STATEMENT: "AND_STATEMENT",
    ASSOCIABLE_RESOURCE: "ASSOCIABLE_RESOURCE",
    ASSOCIATED_RESOURCE_TYPE: "ASSOCIATED_RESOURCE_TYPE",
    ATP_RULE_SET_RESPONSE_INSPECTION: "ATP_RULE_SET_RESPONSE_INSPECTION",
    BODY_PARSING_FALLBACK_BEHAVIOR: "BODY_PARSING_FALLBACK_BEHAVIOR",
    BYTE_MATCH_STATEMENT: "BYTE_MATCH_STATEMENT",
    CHALLENGE_CONFIG: "CHALLENGE_CONFIG",
    CHANGE_PROPAGATION_STATUS: "CHANGE_PROPAGATION_STATUS",
    COOKIE_MATCH_PATTERN: "COOKIE_MATCH_PATTERN",
    CUSTOM_KEYS: "CUSTOM_KEYS",
    CUSTOM_REQUEST_HANDLING: "CUSTOM_REQUEST_HANDLING",
    CUSTOM_RESPONSE: "CUSTOM_RESPONSE",
    CUSTOM_RESPONSE_BODY: "CUSTOM_RESPONSE_BODY",
    DATA_PROTECTION_CONFIG: "DATA_PROTECTION_CONFIG",
    DEFAULT_ACTION: "DEFAULT_ACTION",
    ENTITY_LIMIT: "ENTITY_LIMIT",
    EXCLUDED_RULE: "EXCLUDED_RULE",
    EXPIRE_TIMESTAMP: "EXPIRE_TIMESTAMP",
    FALLBACK_BEHAVIOR: "FALLBACK_BEHAVIOR",
    FIELD_TO_MATCH: "FIELD_TO_MATCH",
    FILTER_CONDITION: "FILTER_CONDITION",
    FIREWALL_MANAGER_STATEMENT: "FIREWALL_MANAGER_STATEMENT",
    FORWARDED_IP_CONFIG: "FORWARDED_IP_CONFIG",
    GEO_MATCH_STATEMENT: "GEO_MATCH_STATEMENT",
    HEADER_MATCH_PATTERN: "HEADER_MATCH_PATTERN",
    HEADER_NAME: "HEADER_NAME",
    IP_ADDRESS: "IP_ADDRESS",
    IP_ADDRESS_VERSION: "IP_ADDRESS_VERSION",
    IP_SET: "IP_SET",
    IP_SET_FORWARDED_IP_CONFIG: "IP_SET_FORWARDED_IP_CONFIG",
    IP_SET_REFERENCE_STATEMENT: "IP_SET_REFERENCE_STATEMENT",
    JSON_MATCH_PATTERN: "JSON_MATCH_PATTERN",
    JSON_MATCH_SCOPE: "JSON_MATCH_SCOPE",
    LABEL_MATCH_STATEMENT: "LABEL_MATCH_STATEMENT",
    LOGGING_FILTER: "LOGGING_FILTER",
    LOG_DESTINATION: "LOG_DESTINATION",
    LOW_REPUTATION_MODE: "LOW_REPUTATION_MODE",
    MANAGED_RULE_GROUP_CONFIG: "MANAGED_RULE_GROUP_CONFIG",
    MANAGED_RULE_SET: "MANAGED_RULE_SET",
    MANAGED_RULE_SET_STATEMENT: "MANAGED_RULE_SET_STATEMENT",
    MAP_MATCH_SCOPE: "MAP_MATCH_SCOPE",
    METRIC_NAME: "METRIC_NAME",
    NOT_STATEMENT: "NOT_STATEMENT",
    OR_STATEMENT: "OR_STATEMENT",
    OVERRIDE_ACTION: "OVERRIDE_ACTION",
    OVERSIZE_HANDLING: "OVERSIZE_HANDLING",
    PAYLOAD_TYPE: "PAYLOAD_TYPE",
    POSITION: "POSITION",
    RATE_BASED_STATEMENT: "RATE_BASED_STATEMENT",
    REGEX_PATTERN_REFERENCE_STATEMENT: "REGEX_PATTERN_REFERENCE_STATEMENT",
    REGEX_PATTERN_SET: "REGEX_PATTERN_SET",
    RESOURCE_ARN: "RESOURCE_ARN",
    RESOURCE_TYPE: "RESOURCE_TYPE",
    RESPONSE_CONTENT_TYPE: "RESPONSE_CONTENT_TYPE",
    RULE: "RULE",
    RULE_ACTION: "RULE_ACTION",
    RULE_GROUP: "RULE_GROUP",
    RULE_GROUP_REFERENCE_STATEMENT: "RULE_GROUP_REFERENCE_STATEMENT",
    SCOPE_DOWN: "SCOPE_DOWN",
    SCOPE_VALUE: "SCOPE_VALUE",
    SINGLE_HEADER: "SINGLE_HEADER",
    SINGLE_QUERY_ARGUMENT: "SINGLE_QUERY_ARGUMENT",
    SIZE_CONSTRAINT_STATEMENT: "SIZE_CONSTRAINT_STATEMENT",
    SQLI_MATCH_STATEMENT: "SQLI_MATCH_STATEMENT",
    STATEMENT: "STATEMENT",
    TAGS: "TAGS",
    TAG_KEYS: "TAG_KEYS",
    TEXT_TRANSFORMATION: "TEXT_TRANSFORMATION",
    TOKEN_DOMAIN: "TOKEN_DOMAIN",
    WEB_ACL: "WEB_ACL",
    XSS_MATCH_STATEMENT: "XSS_MATCH_STATEMENT",
};
const SizeInspectionLimit = {
    KB_16: "KB_16",
    KB_32: "KB_32",
    KB_48: "KB_48",
    KB_64: "KB_64",
};
const Scope = {
    CLOUDFRONT: "CLOUDFRONT",
    REGIONAL: "REGIONAL",
};
const IPAddressVersion = {
    IPV4: "IPV4",
    IPV6: "IPV6",
};
const ResponseContentType = {
    APPLICATION_JSON: "APPLICATION_JSON",
    TEXT_HTML: "TEXT_HTML",
    TEXT_PLAIN: "TEXT_PLAIN",
};
const DataProtectionAction = {
    HASH: "HASH",
    SUBSTITUTION: "SUBSTITUTION",
};
const FieldToProtectType = {
    BODY: "BODY",
    QUERY_STRING: "QUERY_STRING",
    SINGLE_COOKIE: "SINGLE_COOKIE",
    SINGLE_HEADER: "SINGLE_HEADER",
    SINGLE_QUERY_ARGUMENT: "SINGLE_QUERY_ARGUMENT",
};
const LowReputationMode = {
    ACTIVE_UNDER_DDOS: "ACTIVE_UNDER_DDOS",
    ALWAYS_ON: "ALWAYS_ON",
};
const LogScope = {
    CLOUDWATCH_TELEMETRY_RULE_MANAGED: "CLOUDWATCH_TELEMETRY_RULE_MANAGED",
    CUSTOMER: "CUSTOMER",
    SECURITY_LAKE: "SECURITY_LAKE",
};
const LogType = {
    WAF_LOGS: "WAF_LOGS",
};
const Platform = {
    ANDROID: "ANDROID",
    IOS: "IOS",
};
const FilterBehavior = {
    DROP: "DROP",
    KEEP: "KEEP",
};
const FilterRequirement = {
    MEETS_ALL: "MEETS_ALL",
    MEETS_ANY: "MEETS_ANY",
};
const FailureReason = {
    TOKEN_DOMAIN_MISMATCH: "TOKEN_DOMAIN_MISMATCH",
    TOKEN_EXPIRED: "TOKEN_EXPIRED",
    TOKEN_INVALID: "TOKEN_INVALID",
    TOKEN_MISSING: "TOKEN_MISSING",
};
const ResourceType = {
    AMPLIFY: "AMPLIFY",
    API_GATEWAY: "API_GATEWAY",
    APPLICATION_LOAD_BALANCER: "APPLICATION_LOAD_BALANCER",
    APPSYNC: "APPSYNC",
    APP_RUNNER_SERVICE: "APP_RUNNER_SERVICE",
    COGNITIO_USER_POOL: "COGNITO_USER_POOL",
    VERIFIED_ACCESS_INSTANCE: "VERIFIED_ACCESS_INSTANCE",
};

exports.$Command = smithyClient.Command;
exports.__Client = smithyClient.Client;
exports.WAFV2ServiceException = WAFV2ServiceException.WAFV2ServiceException;
exports.ActionValue = ActionValue;
exports.AssociateWebACLCommand = AssociateWebACLCommand;
exports.AssociatedResourceType = AssociatedResourceType;
exports.BodyParsingFallbackBehavior = BodyParsingFallbackBehavior;
exports.CheckCapacityCommand = CheckCapacityCommand;
exports.ComparisonOperator = ComparisonOperator;
exports.CountryCode = CountryCode;
exports.CreateAPIKeyCommand = CreateAPIKeyCommand;
exports.CreateIPSetCommand = CreateIPSetCommand;
exports.CreateRegexPatternSetCommand = CreateRegexPatternSetCommand;
exports.CreateRuleGroupCommand = CreateRuleGroupCommand;
exports.CreateWebACLCommand = CreateWebACLCommand;
exports.DataProtectionAction = DataProtectionAction;
exports.DeleteAPIKeyCommand = DeleteAPIKeyCommand;
exports.DeleteFirewallManagerRuleGroupsCommand = DeleteFirewallManagerRuleGroupsCommand;
exports.DeleteIPSetCommand = DeleteIPSetCommand;
exports.DeleteLoggingConfigurationCommand = DeleteLoggingConfigurationCommand;
exports.DeletePermissionPolicyCommand = DeletePermissionPolicyCommand;
exports.DeleteRegexPatternSetCommand = DeleteRegexPatternSetCommand;
exports.DeleteRuleGroupCommand = DeleteRuleGroupCommand;
exports.DeleteWebACLCommand = DeleteWebACLCommand;
exports.DescribeAllManagedProductsCommand = DescribeAllManagedProductsCommand;
exports.DescribeManagedProductsByVendorCommand = DescribeManagedProductsByVendorCommand;
exports.DescribeManagedRuleGroupCommand = DescribeManagedRuleGroupCommand;
exports.DisassociateWebACLCommand = DisassociateWebACLCommand;
exports.FailureReason = FailureReason;
exports.FallbackBehavior = FallbackBehavior;
exports.FieldToProtectType = FieldToProtectType;
exports.FilterBehavior = FilterBehavior;
exports.FilterRequirement = FilterRequirement;
exports.ForwardedIPPosition = ForwardedIPPosition;
exports.GenerateMobileSdkReleaseUrlCommand = GenerateMobileSdkReleaseUrlCommand;
exports.GetDecryptedAPIKeyCommand = GetDecryptedAPIKeyCommand;
exports.GetIPSetCommand = GetIPSetCommand;
exports.GetLoggingConfigurationCommand = GetLoggingConfigurationCommand;
exports.GetManagedRuleSetCommand = GetManagedRuleSetCommand;
exports.GetMobileSdkReleaseCommand = GetMobileSdkReleaseCommand;
exports.GetPermissionPolicyCommand = GetPermissionPolicyCommand;
exports.GetRateBasedStatementManagedKeysCommand = GetRateBasedStatementManagedKeysCommand;
exports.GetRegexPatternSetCommand = GetRegexPatternSetCommand;
exports.GetRuleGroupCommand = GetRuleGroupCommand;
exports.GetSampledRequestsCommand = GetSampledRequestsCommand;
exports.GetTopPathStatisticsByTrafficCommand = GetTopPathStatisticsByTrafficCommand;
exports.GetWebACLCommand = GetWebACLCommand;
exports.GetWebACLForResourceCommand = GetWebACLForResourceCommand;
exports.IPAddressVersion = IPAddressVersion;
exports.InspectionLevel = InspectionLevel;
exports.JsonMatchScope = JsonMatchScope;
exports.LabelMatchScope = LabelMatchScope;
exports.ListAPIKeysCommand = ListAPIKeysCommand;
exports.ListAvailableManagedRuleGroupVersionsCommand = ListAvailableManagedRuleGroupVersionsCommand;
exports.ListAvailableManagedRuleGroupsCommand = ListAvailableManagedRuleGroupsCommand;
exports.ListIPSetsCommand = ListIPSetsCommand;
exports.ListLoggingConfigurationsCommand = ListLoggingConfigurationsCommand;
exports.ListManagedRuleSetsCommand = ListManagedRuleSetsCommand;
exports.ListMobileSdkReleasesCommand = ListMobileSdkReleasesCommand;
exports.ListRegexPatternSetsCommand = ListRegexPatternSetsCommand;
exports.ListResourcesForWebACLCommand = ListResourcesForWebACLCommand;
exports.ListRuleGroupsCommand = ListRuleGroupsCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListWebACLsCommand = ListWebACLsCommand;
exports.LogScope = LogScope;
exports.LogType = LogType;
exports.LowReputationMode = LowReputationMode;
exports.MapMatchScope = MapMatchScope;
exports.OversizeHandling = OversizeHandling;
exports.ParameterExceptionField = ParameterExceptionField;
exports.PayloadType = PayloadType;
exports.Platform = Platform;
exports.PositionalConstraint = PositionalConstraint;
exports.PutLoggingConfigurationCommand = PutLoggingConfigurationCommand;
exports.PutManagedRuleSetVersionsCommand = PutManagedRuleSetVersionsCommand;
exports.PutPermissionPolicyCommand = PutPermissionPolicyCommand;
exports.RateBasedStatementAggregateKeyType = RateBasedStatementAggregateKeyType;
exports.ResourceType = ResourceType;
exports.ResponseContentType = ResponseContentType;
exports.Scope = Scope;
exports.SensitivityLevel = SensitivityLevel;
exports.SensitivityToAct = SensitivityToAct;
exports.SizeInspectionLimit = SizeInspectionLimit;
exports.TagResourceCommand = TagResourceCommand;
exports.TextTransformationType = TextTransformationType;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateIPSetCommand = UpdateIPSetCommand;
exports.UpdateManagedRuleSetVersionExpiryDateCommand = UpdateManagedRuleSetVersionExpiryDateCommand;
exports.UpdateRegexPatternSetCommand = UpdateRegexPatternSetCommand;
exports.UpdateRuleGroupCommand = UpdateRuleGroupCommand;
exports.UpdateWebACLCommand = UpdateWebACLCommand;
exports.UsageOfAction = UsageOfAction;
exports.WAFV2 = WAFV2;
exports.WAFV2Client = WAFV2Client;
Object.prototype.hasOwnProperty.call(schemas_0, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: schemas_0['__proto__']
    });

Object.keys(schemas_0).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = schemas_0[k];
});
Object.prototype.hasOwnProperty.call(errors, '__proto__') &&
    !Object.prototype.hasOwnProperty.call(exports, '__proto__') &&
    Object.defineProperty(exports, '__proto__', {
        enumerable: true,
        value: errors['__proto__']
    });

Object.keys(errors).forEach(function (k) {
    if (k !== 'default' && !Object.prototype.hasOwnProperty.call(exports, k)) exports[k] = errors[k];
});
