// @aws-sdk/client-wafv2/dist-cjs/models/errors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WAFInvalidPermissionPolicyException = exports.WAFServiceLinkedRoleErrorException = exports.WAFLogDestinationPermissionIssueException = exports.WAFUnsupportedAggregateKeyTypeException = exports.WAFAssociatedItemException = exports.WAFConfigurationWarningException = exports.WAFTagOperationInternalErrorException = exports.WAFTagOperationException = exports.WAFOptimisticLockException = exports.WAFDuplicateItemException = exports.WAFSubscriptionNotFoundException = exports.WAFInvalidResourceException = exports.WAFExpiredManagedRuleGroupVersionException = exports.WAFUnavailableEntityException = exports.WAFNonexistentItemException = exports.WAFLimitsExceededException = exports.WAFInvalidParameterException = exports.WAFInvalidOperationException = exports.WAFInternalErrorException = exports.WAFFeatureNotIncludedInPricingPlanException = void 0;
const WAFV2ServiceException_1 = require("./WAFV2ServiceException");
class WAFFeatureNotIncludedInPricingPlanException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFFeatureNotIncludedInPricingPlanException";
    $fault = "client";
    Message;
    DisallowedFeatures;
    constructor(opts) {
        super({
            name: "WAFFeatureNotIncludedInPricingPlanException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFFeatureNotIncludedInPricingPlanException.prototype);
        this.Message = opts.Message;
        this.DisallowedFeatures = opts.DisallowedFeatures;
    }
}
exports.WAFFeatureNotIncludedInPricingPlanException = WAFFeatureNotIncludedInPricingPlanException;
class WAFInternalErrorException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFInternalErrorException";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "WAFInternalErrorException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFInternalErrorException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFInternalErrorException = WAFInternalErrorException;
class WAFInvalidOperationException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFInvalidOperationException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "WAFInvalidOperationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFInvalidOperationException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFInvalidOperationException = WAFInvalidOperationException;
class WAFInvalidParameterException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFInvalidParameterException";
    $fault = "client";
    Field;
    Parameter;
    Reason;
    constructor(opts) {
        super({
            name: "WAFInvalidParameterException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFInvalidParameterException.prototype);
        this.Field = opts.Field;
        this.Parameter = opts.Parameter;
        this.Reason = opts.Reason;
    }
}
exports.WAFInvalidParameterException = WAFInvalidParameterException;
class WAFLimitsExceededException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFLimitsExceededException";
    $fault = "client";
    Message;
    SourceType;
    constructor(opts) {
        super({
            name: "WAFLimitsExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFLimitsExceededException.prototype);
        this.Message = opts.Message;
        this.SourceType = opts.SourceType;
    }
}
exports.WAFLimitsExceededException = WAFLimitsExceededException;
class WAFNonexistentItemException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFNonexistentItemException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "WAFNonexistentItemException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFNonexistentItemException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFNonexistentItemException = WAFNonexistentItemException;
class WAFUnavailableEntityException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFUnavailableEntityException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "WAFUnavailableEntityException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFUnavailableEntityException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFUnavailableEntityException = WAFUnavailableEntityException;
class WAFExpiredManagedRuleGroupVersionException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFExpiredManagedRuleGroupVersionException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "WAFExpiredManagedRuleGroupVersionException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFExpiredManagedRuleGroupVersionException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFExpiredManagedRuleGroupVersionException = WAFExpiredManagedRuleGroupVersionException;
class WAFInvalidResourceException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFInvalidResourceException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "WAFInvalidResourceException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFInvalidResourceException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFInvalidResourceException = WAFInvalidResourceException;
class WAFSubscriptionNotFoundException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFSubscriptionNotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "WAFSubscriptionNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFSubscriptionNotFoundException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFSubscriptionNotFoundException = WAFSubscriptionNotFoundException;
class WAFDuplicateItemException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFDuplicateItemException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "WAFDuplicateItemException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFDuplicateItemException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFDuplicateItemException = WAFDuplicateItemException;
class WAFOptimisticLockException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFOptimisticLockException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "WAFOptimisticLockException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFOptimisticLockException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFOptimisticLockException = WAFOptimisticLockException;
class WAFTagOperationException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFTagOperationException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "WAFTagOperationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFTagOperationException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFTagOperationException = WAFTagOperationException;
class WAFTagOperationInternalErrorException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFTagOperationInternalErrorException";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "WAFTagOperationInternalErrorException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFTagOperationInternalErrorException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFTagOperationInternalErrorException = WAFTagOperationInternalErrorException;
class WAFConfigurationWarningException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFConfigurationWarningException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "WAFConfigurationWarningException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFConfigurationWarningException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFConfigurationWarningException = WAFConfigurationWarningException;
class WAFAssociatedItemException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFAssociatedItemException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "WAFAssociatedItemException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFAssociatedItemException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFAssociatedItemException = WAFAssociatedItemException;
class WAFUnsupportedAggregateKeyTypeException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFUnsupportedAggregateKeyTypeException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "WAFUnsupportedAggregateKeyTypeException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFUnsupportedAggregateKeyTypeException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFUnsupportedAggregateKeyTypeException = WAFUnsupportedAggregateKeyTypeException;
class WAFLogDestinationPermissionIssueException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFLogDestinationPermissionIssueException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "WAFLogDestinationPermissionIssueException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFLogDestinationPermissionIssueException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFLogDestinationPermissionIssueException = WAFLogDestinationPermissionIssueException;
class WAFServiceLinkedRoleErrorException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFServiceLinkedRoleErrorException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "WAFServiceLinkedRoleErrorException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFServiceLinkedRoleErrorException.prototype);
    }
}
exports.WAFServiceLinkedRoleErrorException = WAFServiceLinkedRoleErrorException;
class WAFInvalidPermissionPolicyException extends WAFV2ServiceException_1.WAFV2ServiceException {
    name = "WAFInvalidPermissionPolicyException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "WAFInvalidPermissionPolicyException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, WAFInvalidPermissionPolicyException.prototype);
        this.Message = opts.Message;
    }
}
exports.WAFInvalidPermissionPolicyException = WAFInvalidPermissionPolicyException;
